### DISCUSSION TITLE
RavenDB Cluster with Docker Compose

### DISCUSSION THREAD

**Discussion by danielmarbach** [ravendb/ravendb#13126] [Q&A]
As part of https://github.com/Particular/NServiceBus.RavenDB/pull/807 I've tried to create a RavenDB Cluster with docker compose for local development. The cluster spins up and is properly licensed with a developer license. The problem is though that when the client connects via `http://localhost:8081`, `http://localhost:8082`, `http://localhost:8083` the node announces the internal IP within docker and then we cannot connect. Technically, we could us the hosts IP and pass it into the compose file as well as the scripts but that seems to work smoothly only with WSL1. With WSL2 the containers are running in their own virtualized networks and require netsh rules on the host

https://docs.microsoft.com/en-us/windows/wsl/networking#accessing-a-wsl-2-distribution-from-your-local-area-network-lan

Any recommendations how to achieve a good local development story that requires a three node cluster?

**Comment by ayende** [comment]
You need to be able to talk between the nodes, and you typically do that using hostnames.

See: 
https://docs.docker.com/compose/compose-file/compose-file-v3/#domainname-hostname-ipc-mac_address-privileged-read_only-shm_size-stdin_open-tty-user-working_dir
